dlt/load/utils.py

```diff
@@ -216,3 +218,44 @@ def _extend_tables_with_table_chain(
                 continue
             result.add(chain_table_name)
     return result
+
+
+def filter_new_jobs(
+    file_names: Sequence[str],
+    capabilities: DestinationCapabilitiesContext,
+    config: LoaderConfiguration,
+) -> Sequence[str]:
+    """Filters the list of new jobs to adhere to max_workers and parallellism strategy"""
+
+    # nothing to do
+    if not file_names:
+        return file_names
+
+    # destination can overwrite ps
+    parallelism_strategy = capabilities.loader_parallelism_strategy or config.parallelism_strategy
```

Comment:
config should always has last say. by default config should be None - no strategy should be forced

Reply:
fixed